File: phase-3/Hash Maps/hashing.cpp
#include<bits/stdc++.h>
using namespace std;

int main(){
    int n;
    cout << "Take the length of array: ";
    cin >> n;
    int arr[n];
    for(int i = 0; i < n; i++){
        cin >> arr[i];
    }

    // Precompute
    int hash[13] = {0};
    for (int i = 0; i < n; i++)
    {
        hash[arr[i]] += 1;
    }

    int q;
    cout << "Take the number of queries: ";
    cin >> q;
    while (q--){
        int x;
        cout << "Enter the element to be searched: ";
        cin >> x;
        
        // Fetch
        if (x < n && x >= 0){
        cout << "Frequency of " << x << " is " << hash[x] << endl;
        }
        else{
        cout << "Frequency of " << x << " is 0" << endl;
    }
}
}

File: phase-3/Hash Maps/hashing.py
n = int(input("Take the length of array: "))
arr = []
for i in range(n):
    arr.append(int(input()))

# Precompute
hash_map = [0] * n
for i in range(n):
    hash_map[arr[i]] += 1

q = int(input("Take the number of queries: "))
while q > 0:
    x = int(input("Enter the element to be searched: "))
    
    # Fetch
    if x < n:
        print(f"Frequency of {x} is {hash_map[x]}")
    else:
        print(f"Frequency of {x} is 0")
    
    q -= 1
